singlelayerqg.jl

```diff
@@ -1,15 +1,17 @@
-module BarotropicQG
+module SingleLayerQG
 
 export
   Problem,
   set_zeta!,
   updatevars!,
 
-  energy,
```

Comment:
Does it make sense to have `energy()` function giving the total energy of the flow? For a purely barotropic setup that would be simply the kinetic energy, when the deformation wavenumber is non-zero then energy would be the sum of kinetic + potential energy. We can do this by having some `AbstractTypes` for `Params`, e.g., something like

```Julia
energy(prob) = energy(prob.vars, prob.params, prob.grid)
energy(vars, params::BarotropicParams, grid) = kinetic_energy(vars, params, grid)
energy(vars, params::EquivalentBarotropicParams, grid) = kinetic_energy(vars, params, grid) + potential_energy(vars, params, grid)
```

Reply:
(All previously `BarotropicQG` examples fail to run because they are calling `SingleLayer.energy` function which does not exist... 5095b3d added a temp fix to that.)

Reply:
resolved with 4650164 and 049bad0